types/parameterized_single_integer_param_type.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/types/leaf_parameters"
+)
+
+// parameterizedTypeSingleIntegerParam This is a generic type to represent parameterized type with a single integer parameter
+type parameterizedTypeSingleIntegerParam[T VarCharType | FixedCharType | FixedBinaryType | PrecisionTimestampType | PrecisionTimestampTzType] struct {
```

Comment:
I feel like the constraint is kind of pointless since the type is not exported.

Reply:
yes, not really required to strict export. so removed types as constraint but added interface as constraint to simplify baseString as mentioned in [this](https://github.com/substrait-io/substrait-go/pull/52#discussion_r1752303432) comment